src/main/java/org/folio/circulation/resources/LoanCirculationRulesEngineResource.java

```diff
@@ -0,0 +1,79 @@
+package org.folio.circulation.resources;
```

Comment:
Thinking more about LoanCirculationRulesEngineResource and RequestCirculationRulesEngineResource, we can see that these classes couple the rules checking functionality with the handling of service requests.  Consequently, within mod-circulation itself (e.g., LoanPolicyRepository.lookupLoanPolicyId()) when there is a need to apply the rules it has to use CirculationRulesClient to call LoanCirculationRulesEngineResource.apply(), which means that it has to go out of mod-circulation, to Okapi, and back into itself.  This is wildly inefficient.  I'd like to suggest that we decouple the API service requests handling from the rules lookup/apply functionality so that classes like LoanPolicyRepository doesn't have to use CirculationRulesClient.  Effectively we will have a new class that extends AbstractCirculationRulesEngineResource to have the rules lookup/apply functionality. LoanCirculationRulesEngineResource  will call it in the apply() methods, passing appropriate parameters from the routingContext to apply the rules. Now internal code such as LoanPolicyRepository can also call this new class without going out to Okapi.

Reply:
When the original work was done to determine the loan policy using the circulation rules, @skoczko specifically requested that the rules be applied via a round trip back to mod-circulation.

I've raised [CIRC-239](https://issues.folio.org/browse/CIRC-239) to further discuss this, as I think it is out of scope for this change.

Reply:
Even though it's out of scope I think it has serious performance implications.  If we don't decide on this one way or another it will become technical debt that we might forget to revisit.

Reply:
@mtraneis This is being actively discussed at the moment, I only don't want to block downstream work on this change